sql/src/main/java/io/crate/execution/jobs/DistResultRXTask.java

```diff
@@ -183,9 +147,16 @@ private void triggerConsumer() {
         }
     }
 
-    private void mergeAndTriggerConsumer() {
+    private void merge(Map<Integer, Bucket> bucketsByIdx) {
         try {
-            mergeBuckets();
+            List<KeyIterable<Integer, Row>> buckets = new ArrayList<>(numBuckets);
+            synchronized (lock) {
```

Comment:
this won't fix concurrent issues as the MergeBucketReceiver is locking on a different instance